src/components/BackToTopButton/index.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { BackToTopStyles } from './index.styles';
+
+export default class BackToTopButton extends Component {
+  backToTopButton: any;
+  constructor(props: any) {
+    super(props);
+    this.backToTopButton = React.createRef();
+  }
+
+  componentDidMount() {
+    document.addEventListener('scroll', () => {
+      if (window.scrollY > 1000) {
+        this.backToTopButton.current.classList.add('open');
+      } else {
+        this.backToTopButton.current.classList.remove('open');
+      }
+    });
+
+    document.addEventListener('click', () => {
```

Comment:
This should be a function on the component in JSX:
```javascript
        <BackToTopStyles
          onClick={() => window.scrollTo(0, 0)}
          ...
        >
```

Reply:
😳  riiight... Silly mistake.